pubsub/iam.js

```diff
@@ -0,0 +1,183 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+var async = require('async');
+var utils = require('./subscription');
+var createTopic = utils.createTopic;
+var subscribe = utils.subscribe;
+
+// [START get_topic_policy]
+function getTopicPolicy(topic, callback) {
+  // Retrieve the IAM policy for the provided topic
+  topic.iam.getPolicy(callback);
+}
+// [END get_topic_policy]
+
+// [START get_subscription_policy]
+function getSubscriptionPolicy(subscription, callback) {
+  // Retrieve the IAM policy for the provided subscription
+  subscription.iam.getPolicy(callback);
+}
+// [END get_subscription_policy]
+
+// [START set_topic_policy]
+function setTopicPolicy(topic, callback) {
+  // Policy update
+  var myPolicy = {
+    bindings: [
+      {
+        role: 'roles/pubsub.subscriber',
+        members: ['serviceAccount:myotherproject@appspot.gserviceaccount.com']
+      }
+    ]
+  };
+
+  // Retrieve the IAM policy for the provided topic
+  topic.iam.setPolicy(myPolicy, callback);
+}
+// [END set_topic_policy]
+
+// [START set_subscription_policy]
+function setSubscriptionPolicy(subscription, callback) {
+  // Policy update
+  var myPolicy = {
+    bindings: [
+      {
+        role: 'roles/pubsub.subscriber',
+        members: ['serviceAccount:myotherproject@appspot.gserviceaccount.com']
+      }
+    ]
+  };
+
+  // Retrieve the IAM policy for the provided subscription
+  subscription.iam.setPolicy(myPolicy, callback);
+}
+// [END set_subscription_policy]
+
+// [START test_topic_permissions]
+function testTopicPermissions(topic, callback) {
+  var tests = [
+    'pubsub.topics.attachSubscription',
+    'pubsub.topics.publish',
+    'pubsub.topics.update'
+  ];
+
+  // Retrieve the IAM policy for the provided topic
+  topic.iam.testPermissions(tests, callback);
+}
+// [END test_topic_permissions]
+
+// [START test_subscription_permissions]
+function testSubscriptionPermissions(subscription, callback) {
+  var tests = [
+    'pubsub.subscriptions.consume',
+    'pubsub.subscriptions.update'
+  ];
+
+  // Retrieve the IAM policy for the provided subscription
+  subscription.iam.testPermissions(tests, callback);
+}
+// [END test_subscription_permissions]
+
+exports.getTopicPolicy = getTopicPolicy;
+exports.getSubscriptionPolicy = getSubscriptionPolicy;
+exports.setTopicPolicy = setTopicPolicy;
+exports.setSubscriptionPolicy = setSubscriptionPolicy;
+exports.testTopicPermissions = testTopicPermissions;
+exports.testSubscriptionPermissions = testSubscriptionPermissions;
+exports.runSample = runSample;
+
+function runSample(callback) {
+  var _subscription;
+  var _topic;
+  // Gather responses
+  var responses = [];
+  async.waterfall([
```

Comment:
I personally find this part really hard to follow. I don't know if you plan on showing it in the docs.